internal/kms/testing.go

```diff
@@ -2,276 +2,38 @@ package kms
-	require.NotEmpty(rootKeyVersionId)
-	k, err := NewDatabaseKeyVersion(databaseKeyId, key, rootKeyVersionId)
-	require.NoError(err)
-	id, err := newDatabaseKeyVersionId()
-	require.NoError(err)
-	k.PrivateId = id
-	err = k.Encrypt(context.Background(), rootKeyVersionWrapper)
-	require.NoError(err)
-	err = rw.Create(context.Background(), k)
-	require.NoError(err)
-	return k
-}
-
-func TestOplogKey(t testing.TB, conn *db.DB, rootKeyId string) *OplogKey {
-	t.Helper()
-	require := require.New(t)
-	rw := db.New(conn)
-	db.TestDeleteWhere(t, conn, func() interface{} { i := AllocOplogKey(); return &i }(), "root_key_id = ?", rootKeyId)
-	k, err := NewOplogKey(rootKeyId)
-	require.NoError(err)
-	id, err := newOplogKeyId()
-	require.NoError(err)
-	k.PrivateId = id
-	k.RootKeyId = rootKeyId
-	err = rw.Create(context.Background(), k)
-	require.NoError(err)
-	return k
-}
-
-func TestOplogKeyVersion(t testing.TB, conn *db.DB, rootKeyVersionWrapper wrapping.Wrapper, oplogKeyId string, key []byte) *OplogKeyVersion {
-	t.Helper()
-	require := require.New(t)
-	rw := db.New(conn)
-	rootKeyId, err := rootKeyVersionWrapper.KeyId(context.Background())
-	require.NoError(err)
-	rootKeyVersionId := rootKeyId
-	require.NotEmpty(rootKeyVersionId)
-	k, err := NewOplogKeyVersion(oplogKeyId, key, rootKeyVersionId)
-	require.NoError(err)
-	id, err := newOplogKeyVersionId()
-	require.NoError(err)
-	k.PrivateId = id
-	err = k.Encrypt(context.Background(), rootKeyVersionWrapper)
-	require.NoError(err)
-	err = rw.Create(context.Background(), k)
-	require.NoError(err)
-	return k
-}
-
-func TestTokenKey(t testing.TB, conn *db.DB, rootKeyId string) *TokenKey {
-	t.Helper()
-	require := require.New(t)
-	db.TestDeleteWhere(t, conn, func() interface{} { i := AllocTokenKey(); return &i }(), "root_key_id = ?", rootKeyId)
-	rw := db.New(conn)
-	k, err := NewTokenKey(rootKeyId)
+	_, err = kms.underlying.GetExternalRootWrapper()
 	require.NoError(err)
-	id, err := newTokenKeyId()
-	require.NoError(err)
-	k.PrivateId = id
-	k.RootKeyId = rootKeyId
-	err = rw.Create(context.Background(), k)
-	require.NoError(err)
-	return k
+	return kms
 }
 
-func TestTokenKeyVersion(t testing.TB, conn *db.DB, rootKeyVersionWrapper wrapping.Wrapper, tokenKeyId string, key []byte) *TokenKeyVersion {
-	t.Helper()
-	require := require.New(t)
-	rw := db.New(conn)
-	rootKeyId, err := rootKeyVersionWrapper.KeyId(context.Background())
-	require.NoError(err)
-	rootKeyVersionId := rootKeyId
-	require.NotEmpty(rootKeyVersionId)
-	k, err := NewTokenKeyVersion(tokenKeyId, key, rootKeyVersionId)
-	require.NoError(err)
-	id, err := newTokenKeyVersionId()
-	require.NoError(err)
-	k.PrivateId = id
-	err = k.Encrypt(context.Background(), rootKeyVersionWrapper)
-	require.NoError(err)
-	err = rw.Create(context.Background(), k)
-	require.NoError(err)
-	return k
+// TestKmsDeleteKeyPurpose allows you to delete a KeyPurpose for testing.
+func TestKmsDeleteKeyPurpose(t *testing.T, conn *db.DB, purpose KeyPurpose) {
```

Comment:
Mind making these new functions use `testing.TB` like the rest of the helpers?

Reply:
happy to.  ty!